nixos/modules/services/games/minecraft-server.nix

```diff
@@ -61,13 +286,20 @@ in
       serviceConfig.User    = "minecraft";
       script = ''
         cd ${cfg.dataDir}
+
+        cp ${eulaFile} eula.txt
+        cp ${whitelistFile} whitelist.json
```

Comment:
This might overwrite existing files that were put into place before the service change.

Reply:
It absolutely overwrites the files. Maybe a nicer approach would be to copy the originals to `<filename>.old`?

In my opinion `eula.txt` makes sense to always write by default since it allows the server to start automatically without any configuration and its really only writing `eula=true`.  The current default does not start without agreeing to the eula.

However I agree that for `whitelist.json` and `server.properties`  overwriting may not be the best first choice. I could change the option whitelist to have a default value of null in which case `whitelist.json` will not be written. 

 `server.properties` is a bit trickier. This is because `server.properties` has some networking configuration server ip, minecraft port, enable query, query port, enable rcon, rcon port, rcon password. These ports are needed to enable the correct firewall settings. These settings are what NixOS was designed for thus I really believe that NixOS should be controlling this. So im split on this file and how to correctly handle it. What are your thoughts?

Reply:
In general, i learned right now, it is better to force symlink those configuration files to the ```/nix/store``` because then it is clear that the state of these files is managed by nix. (Given the config files do not need to be writable by the minecraft server, of course)

And for notifying anybody who attempts to update the service to the managed config version, one could fail the service if the file exists and is not a symlink already. (which is a bad hack but i can't think of a better idea right now)

Reply:
So I will need to double check but I know that Minecraft requires that `server.properties` is writable. I was running into issues when I symlinked all of the files. I'll get back to you on that

Reply:
I also forgot that the configuration files allow for comments with `# ...` so that would be a way of signaling that `server.properties` is managed by nixos

Reply:
It seems the ```server.properties``` only needs to be writable if it's missing or incomplete.

Taken from https://minecraft.gamepedia.com/Server.properties
If the server.properties file does not list all properties (such as if a new version of the server adds new properties, or if the file does not exist), then on startup the server will rewrite the server.properties file with the new properties listed and set them to their default values.

So maybe there are yet to be documented properties for the 1.13 version, that were missing?